From 931288137360b4790a5ac5e9cc7435012c3c8ec4 Mon Sep 17 00:00:00 2001
From: "Sodhi, Vunny" <vunny.sodhi@intel.com>
Date: Tue, 26 Mar 2019 09:59:21 +0800
Subject: [PATCH 1080/1214] drm/i915: Always enable ARAT irq and GuC redirect

Always enable ARAT irq so that grace period expiry handled properly
in GuC to avoid session teardown for valid configuration.

Change-Id: Ic325519a251fb15018fda2e2bb7e3a7eb6dc7d28
Tracked-On: OAM-77673
Signed-off-by: Sodhi, Vunny <vunny.sodhi@intel.com>
---
 drivers/gpu/drm/i915/i915_irq.c | 6 ++++--
 1 file changed, 4 insertions(+), 2 deletions(-)

diff --git a/drivers/gpu/drm/i915/i915_irq.c b/drivers/gpu/drm/i915/i915_irq.c
index 6aec41b..6294ce5 100644
--- a/drivers/gpu/drm/i915/i915_irq.c
+++ b/drivers/gpu/drm/i915/i915_irq.c
@@ -4840,8 +4840,10 @@ void intel_irq_init(struct drm_i915_private *dev_priv)
 	if (INTEL_GEN(dev_priv) <= 7)
 		rps->pm_intrmsk_mbz |= GEN6_PM_RP_UP_EI_EXPIRED;
 
-	if (INTEL_GEN(dev_priv) >= 8)
-		rps->pm_intrmsk_mbz |= GEN8_PMINTR_DISABLE_REDIRECT_TO_GUC;
+	if (INTEL_GEN(dev_priv) >= 8) {
+		rps->pm_intrmsk_mbz |= ARAT_EXPIRED_INTRMSK;
+		rps->pm_intrmsk_mbz &= ~GEN8_PMINTR_DISABLE_REDIRECT_TO_GUC;
+	}
 
 	if (IS_GEN2(dev_priv)) {
 		/* Gen2 doesn't have a hardware frame counter */
-- 
2.7.4

